cmd/infra/powervs/destroy.go

```diff
@@ -202,31 +230,37 @@ func (options *DestroyInfraOptions) DestroyInfra(ctx context.Context, infra *Inf
 	if !skipPowerVs {
 		session, err = createPowerVSSession(accountID, options.Region, options.Zone, options.Debug)
 		if err != nil {
-			return err
+			errL = append(errL, fmt.Errorf("error creating powervs session: %w", err))
+			log(options.InfraID).Error(err, "error creating powervs session")
```

Comment:
Why do we even continue if `createPowerVSSession` fails? may be we need to create a function for this so that we can safely return instead of continuing

Reply:
Refactored the code to use separate func for destroying resources in powervs and vpc. ptal!